xbmc/pvr/epg/EpgDatabase.cpp

```diff
@@ -990,6 +988,32 @@ std::vector<std::shared_ptr<CPVREpgInfoTag>> CPVREpgDatabase::GetAllEpgTags(int
   return {};
 }
 
+std::vector<std::string> CPVREpgDatabase::GetAllIconPaths(int iEpgID)
+{
+  CSingleLock lock(m_critSection);
+  const std::string strQuery =
+      PrepareSQL("SELECT sIconPath FROM epgtags WHERE idEpg = %u;", iEpgID);
+  if (ResultQuery(strQuery))
+  {
+    try
+    {
+      std::vector<std::string> paths;
+      while (!m_pDS->eof())
+      {
+        paths.emplace_back(m_pDS->fv("sIconPath").get_asString());
+        m_pDS->next();
+      }
+      m_pDS->close();
+      return paths;
+    }
+    catch (...)
+    {
+      CLog::LogF(LOGERROR, "Could not load tags for EPG ({})", iEpgID);
+    }
+  }
```

Comment:
Is this a standard pattern for DB reads? If not, catch everything makes me worry we're not sure what could go wrong.

Reply:
Yes, it's like this everywhere in PVR and EPG database. Introduced long before my Kodi times.

Reply:
Ok, cool.